// angular2-swapi/esm5/lib/angular2-swapi.models.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * @record
 */
export function People() { }
function People_tsickle_Closure_declarations() {
    /** @type {?} */
    People.prototype.name;
    /** @type {?} */
    People.prototype.birth_year;
    /** @type {?} */
    People.prototype.eye_color;
    /** @type {?} */
    People.prototype.gender;
    /** @type {?} */
    People.prototype.hair_color;
    /** @type {?} */
    People.prototype.height;
    /** @type {?} */
    People.prototype.mass;
    /** @type {?} */
    People.prototype.skin_color;
    /** @type {?} */
    People.prototype.homeworld;
    /** @type {?} */
    People.prototype.films;
    /** @type {?} */
    People.prototype.species;
    /** @type {?} */
    People.prototype.starships;
    /** @type {?} */
    People.prototype.vehicles;
    /** @type {?} */
    People.prototype.url;
    /** @type {?} */
    People.prototype.created;
    /** @type {?} */
    People.prototype.edited;
}
/**
 * @record
 */
export function Film() { }
function Film_tsickle_Closure_declarations() {
    /** @type {?} */
    Film.prototype.title;
    /** @type {?} */
    Film.prototype.episode_id;
    /** @type {?} */
    Film.prototype.opening_crawl;
    /** @type {?} */
    Film.prototype.director;
    /** @type {?} */
    Film.prototype.producer;
    /** @type {?} */
    Film.prototype.release_date;
    /** @type {?} */
    Film.prototype.species;
    /** @type {?} */
    Film.prototype.starships;
    /** @type {?} */
    Film.prototype.vehicles;
    /** @type {?} */
    Film.prototype.characters;
    /** @type {?} */
    Film.prototype.planets;
    /** @type {?} */
    Film.prototype.url;
    /** @type {?} */
    Film.prototype.created;
    /** @type {?} */
    Film.prototype.edited;
}
/**
 * @record
 */
export function Starship() { }
function Starship_tsickle_Closure_declarations() {
    /** @type {?} */
    Starship.prototype.name;
    /** @type {?} */
    Starship.prototype.model;
    /** @type {?} */
    Starship.prototype.starship_class;
    /** @type {?} */
    Starship.prototype.manufacturer;
    /** @type {?} */
    Starship.prototype.cost_in_credits;
    /** @type {?} */
    Starship.prototype.length;
    /** @type {?} */
    Starship.prototype.crew;
    /** @type {?} */
    Starship.prototype.passengers;
    /** @type {?} */
    Starship.prototype.max_atmosphering_speed;
    /** @type {?} */
    Starship.prototype.hyperdrive_rating;
    /** @type {?} */
    Starship.prototype.MGLT;
    /** @type {?} */
    Starship.prototype.cargo_capacity;
    /** @type {?} */
    Starship.prototype.consumables;
    /** @type {?} */
    Starship.prototype.films;
    /** @type {?} */
    Starship.prototype.pilots;
    /** @type {?} */
    Starship.prototype.url;
    /** @type {?} */
    Starship.prototype.created;
    /** @type {?} */
    Starship.prototype.edited;
}
/**
 * @record
 */
export function Vehicle() { }
function Vehicle_tsickle_Closure_declarations() {
    /** @type {?} */
    Vehicle.prototype.name;
    /** @type {?} */
    Vehicle.prototype.model;
    /** @type {?} */
    Vehicle.prototype.vehicle_class;
    /** @type {?} */
    Vehicle.prototype.manufacturer;
    /** @type {?} */
    Vehicle.prototype.length;
    /** @type {?} */
    Vehicle.prototype.cost_in_credits;
    /** @type {?} */
    Vehicle.prototype.crew;
    /** @type {?} */
    Vehicle.prototype.passengers;
    /** @type {?} */
    Vehicle.prototype.max_atmosphering_speed;
    /** @type {?} */
    Vehicle.prototype.cargo_capacity;
    /** @type {?} */
    Vehicle.prototype.consumables;
    /** @type {?} */
    Vehicle.prototype.films;
    /** @type {?} */
    Vehicle.prototype.pilots;
    /** @type {?} */
    Vehicle.prototype.url;
    /** @type {?} */
    Vehicle.prototype.created;
    /** @type {?} */
    Vehicle.prototype.edited;
}
/**
 * @record
 */
export function Species() { }
function Species_tsickle_Closure_declarations() {
    /** @type {?} */
    Species.prototype.name;
    /** @type {?} */
    Species.prototype.classification;
    /** @type {?} */
    Species.prototype.designation;
    /** @type {?} */
    Species.prototype.average_height;
    /** @type {?} */
    Species.prototype.average_lifespan;
    /** @type {?} */
    Species.prototype.eye_colors;
    /** @type {?} */
    Species.prototype.hair_colors;
    /** @type {?} */
    Species.prototype.skin_colors;
    /** @type {?} */
    Species.prototype.language;
    /** @type {?} */
    Species.prototype.homeworld;
    /** @type {?} */
    Species.prototype.people;
    /** @type {?} */
    Species.prototype.films;
    /** @type {?} */
    Species.prototype.url;
    /** @type {?} */
    Species.prototype.created;
    /** @type {?} */
    Species.prototype.edited;
}
/**
 * @record
 */
export function Planet() { }
function Planet_tsickle_Closure_declarations() {
    /** @type {?} */
    Planet.prototype.name;
    /** @type {?} */
    Planet.prototype.diameter;
    /** @type {?} */
    Planet.prototype.rotation_period;
    /** @type {?} */
    Planet.prototype.orbital_period;
    /** @type {?} */
    Planet.prototype.gravity;
    /** @type {?} */
    Planet.prototype.population;
    /** @type {?} */
    Planet.prototype.climate;
    /** @type {?} */
    Planet.prototype.terrain;
    /** @type {?} */
    Planet.prototype.surface_water;
    /** @type {?} */
    Planet.prototype.residents;
    /** @type {?} */
    Planet.prototype.films;
    /** @type {?} */
    Planet.prototype.url;
    /** @type {?} */
    Planet.prototype.created;
    /** @type {?} */
    Planet.prototype.edited;
}
